extends CharacterBody2D
class_name Player

const GRAVITY = 500.0
const WALK_SPEED = 200
const JUMP_FORCE = -330.0

var active_mask: Mask
var mask_sprite: Sprite2D

func switch_mask(mask: Mask) -> void:
	active_mask = mask
	active_mask.connect("start_mask_rotation", rotate_mask)

func rotate_mask(left: bool) -> void:
	var tween = create_tween()
	if left:
		tween.tween_property($Mask, "rotation_degrees", active_mask.rotation_degrees - 90, 0.2)
	else:
		tween.tween_property($Mask, "rotation_degrees", active_mask.rotation_degrees + 90, 0.2)

func _ready() -> void:
	mask_sprite = $Mask
	scale *= 4

func _input(event: InputEvent) -> void:
	if event.is_action_pressed("jump"):
		jump()

func jump() -> void:
	if is_on_floor():
		self.velocity.y = JUMP_FORCE

func _physics_process(delta):
	velocity.y += delta * GRAVITY

	if Input.is_action_pressed("ui_left"):
		velocity.x = -WALK_SPEED
	elif Input.is_action_pressed("ui_right"):
		velocity.x =  WALK_SPEED
	else:
		velocity.x = 0

	move_and_slide()
